src/civetweb.c

```diff
@@ -14345,6 +14345,36 @@ set_ssl_option(struct mg_context *ctx)
 		return 1;
 	}
 
+	/* Check for external SSL_CTX */
+	SSL_CTX* ssl_ctx = 0;
+	callback_ret =
+	    (ctx->callbacks.get_external_ssl_ctx == NULL)
+	        ? 0
+	        : (ctx->callbacks.get_external_ssl_ctx(&ssl_ctx, ctx->user_data));
+
+	if (callback_ret < 0) {
+		mg_cry(fc(ctx), "get_external_ssl_ctx callback returned error: %i", callback_ret);
+		return 0;
+	}
+	else if (callback_ret > 0) {
+		ctx->ssl_ctx = ssl_ctx;
+		if (!initialize_ssl(ebuf, sizeof(ebuf))) {
+	 	   mg_cry(fc(ctx), "%s", ebuf);
+		   return 0;
+	    }
+#if !defined(NO_SSL_DL)
+		if (!ssllib_dll_handle) {
+			ssllib_dll_handle = load_dll(ebuf, sizeof(ebuf), SSL_LIB, ssl_sw);
+			if (!ssllib_dll_handle) {
+				mg_cry(fc(ctx), "%s", ebuf);
+				return 0;
+			}
+		}
+#endif /* NO_SSL_DL */
```

Comment:
Review comment - block `#if !defined(NO_SSL_DL)`:
This is already done in `initialize_ssl` in the current version.
TODO: Verify in test.